hyperspy/_signals/signal1d.py

```diff
@@ -454,9 +454,8 @@ def shift1D(
                 post_array = da.full(tuple(post_shape),
                                      fill_value,
                                      chunks=tuple(post_chunks))
-
                 self.data = da.concatenate((pre_array, self.data, post_array),
-                                           axis=ind)
+                                           axis=ind).rechunk({ind:-1})
```

Comment:
is rechunking strictly required? If not, I would suggest to add a keyword to control whether to automatically rechunk.

Reply:
If the code runs as it does right now then yes it is required because you are changing the size of the array and your chunk size no longer spans your signal array.  That should never be the case, but I should really add in a line of code to catch that case.

For this function I can move the padding inside the function (It probably should be anyways) and then you don't have to deal with the rechunking.

Reply:
Sounds good.